deployment/changeset.go

```diff
@@ -20,12 +20,22 @@ var (
 // If the configuration is unexpected type or format, the changeset should return ErrInvalidConfig.
 type ChangeSet[C any] func(e Environment, config C) (ChangesetOutput, error)
 
+// ProposedJob represents a job spec which has been proposed to a node, with the JobId returned by the
+// Job Distributor.
+type ProposedJob struct {
+	JobId string
+	Node  string
```

Comment:
is this the NodeId returned by JD?

Reply:
It's whatever node was going to be used in the existing code - I believe JobSpec right now is indexed by node. That one.